### PR TITLE
chore: Convert export to language to the new config COMPASS-6207

### PR DESCRIPTION
**Pull request by lerouxb**
This just switches compass-export-to-language over to the new shared config. I then did the minimal amount of work necessary to get check and test to pass.

### CODE REVIEW

File: configs/mocha-config-compass/register/jsdom-global-register.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 if (typeof window === 'undefined') {
-  require('global-jsdom')(undefined, { pretendToBeVisual: true });
+  require('global-jsdom')(undefined, { pretendToBeVisual: true, runScripts: 'dangerously' });
+  console.log('window', window, window === 'undefined');
```
Comment (lerouxb) [inline configs/mocha-config-compass/register/jsdom-global-register.js:3]: It looks fine here, btw. But who knows which `window` object leafygreen sees.

File: packages/bson-transpilers/codegeneration/javascript/Visitor.js [hunk 2]
```diff
@@ -604,9 +604,7 @@ module.exports = (CodeGenerationVisitor) => class Visitor extends CodeGeneration
       Buffer: Buffer,
       __result: {}
     };
-    const ctx = new Context(sandbox);
-    const res = ctx.evaluate('__result = ' + input);
-    ctx.destroy();
+    const res = vm.runInContext('__result = ' + input, vm.createContext(sandbox));
```
Comment (lerouxb) [inline packages/bson-transpilers/codegeneration/javascript/Visitor.js:607]: On node this is all that [context-eval](https://github.com/amasad/context-eval) does. For the browser (ie. data explorer) the plan is to use a [polyfill](https://github.com/browserify/vm-browserify).

context-eval broke for us in this upgrade for kinda tricky reasons:

Before this package was using jsdom via [jsdom-global](https://github.com/rstacruz/jsdom-global) which has a [hardcoded list of properties](https://github.com/rstacruz/jsdom-global/blob/master/keys.js) to copy over to the global (window) object. The shared config (ie. the new way) uses [global-jsdom](https://github.com/modosc/global-jsdom) which uses [Object.getOwnPropertyNames(window)](https://github.com/modosc/global-jsdom/blob/5b90b116426869054cfd551daa7d83b733a35395/commonjs/index.js#L38).

For whatever reason bson-transpiler executes some of the javascript it transpiles and it does this in a context. That requires the jsdom `runScripts: "dangerously"` option to be set. Unfortunately this causes (for reasons we're not 100% clear on, I think) `Object.getOwnPropertyNames(window)` to not return things like `Element` which breaks leafygreen.

So to recap: Without runScripts: "dangeriously" tests would fail with ObjectId not defined. With it leafygreen would fail.

Looking at context-eval, it looks for the existance of the `window` object to know if it is running in node (where it can use the vm module) or in the browser (where it falls back to using an iframe). Since the tests are running in node could just be using `vm`. In electron we should have access to it as well. Elsewhere we shouldn't probably be polyfilling it.

So the cleanest solution seems to be to just not use context-eval and polyfill node's vm module if necessary.

File: packages/compass/package.json [hunk 3]
```diff
@@ -182,7 +182,7 @@
     "@mongodb-js/compass-database": "^3.2.0",
     "@mongodb-js/compass-databases-collections": "^1.2.0",
     "@mongodb-js/compass-explain-plan": "^6.2.0",
-    "@mongodb-js/compass-export-to-language": "^8.2.0",
+    "@mongodb-js/compass-export-to-language": "^0.1.0",
```
Comment (gribnoysup) [inline packages/compass/package.json:185]: We probably want the old version to be preserved

Comment (lerouxb) [reply packages/compass/package.json:185]: aah good point. I think I miss that every time :)

Comment (gribnoysup) [reply packages/compass/package.json:185]: Yeah, trips me up all the time too 😅

**Review by addaleax — APPROVED** [review]

File: packages/compass-export-to-language/src/modules/count-aggregation-stages-in-string.js [hunk 4]
```diff
@@ -13,17 +13,33 @@ export function countAggregationStagesInString(str) {
   if (!dummySandbox) {
     dummySandbox = vm.createContext(Object.create(null), {
       codeGeneration: { strings: false, wasm: false },
-      microtaskMode: 'afterEvaluate'
+      microtaskMode: 'afterEvaluate',
     });
-    vm.runInContext([
```
Comment (lerouxb) [inline packages/compass-export-to-language/src/modules/count-aggregation-stages-in-string.js:18]: I just noticed afterwards that here we were already using vm.runInContext directly anyway.

Comment (addaleax) [reply packages/compass-export-to-language/src/modules/count-aggregation-stages-in-string.js:18]: … to be fair, I only used this because it was quick and reliable. This is not a good long-term solution! :)

Comment (lerouxb) [reply packages/compass-export-to-language/src/modules/count-aggregation-stages-in-string.js:18]: Aaah now I remember what this was. OK so probably not releveant, then. Couldn't tell from the diff.

**Comment by lerouxb** [comment]
From mms's perspective they can just change this line https://github.com/10gen/mms/blob/c5b032ada223f54e4eabba79a1e10d5bf6e5c5bb/client/scripts/webpack/createWebpackConfigs.js#L196 to `vm: 'vm-browserify'`

(via https://mongodb.slack.com/archives/G2L10JAV7/p1666882029693469?thread_ts=1666881513.720759&cid=G2L10JAV7)

File: packages/bson-transpilers/codegeneration/javascript/Visitor.js (hunk 2 above)
Comment (mcasimir) [reply packages/bson-transpilers/codegeneration/javascript/Visitor.js:607]: I may be missing something do we already have the polyfill in this PR? Does something change for Data Explorer?

Comment (lerouxb) [reply packages/bson-transpilers/codegeneration/javascript/Visitor.js:607]: We don't seem to need the polyfill at this point because electron seems to make it work for us. Data Explorer will have to add it. We're testing to see if it will work.

Btw we might eventually need the polyfill as well if we make the changes to not rely on electron and use the same stuff they do.

**Comment by gribnoysup** [comment]
Just checked with @lerouxb and this works with a polyfill correctly:

image
